nvm.sh

```diff
@@ -2377,14 +2442,19 @@ nvm() {
             shift
           fi
         else
-          nvm_rc_version
-          if [ $version_not_provided -eq 1 ] && [ -z "$NVM_RC_VERSION" ]; then
+          nvm_rc_version || nvm_package_json_version
+          if [ -n "$NVM_RC_VERSION" ]; then
```

Comment:
all these vars should use `${FOO-}` syntax so that the `-x` option won't error out on them (at least until it's ensured that they're set)